### PR TITLE
fix(agw): updated env.j2 file

### PR DESCRIPTION
**Pull request by ShubhamTatvamasi**
Signed-off-by: Shubham Tatvamasi <shubhamtatvamasi@gmail.com>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/deploy/roles/agw_docker/tasks/main.yml [hunk 1]
```diff
@@ -63,7 +63,7 @@
   template:
     src: env.j2
     dest: /var/opt/magma/docker/.env
-    mode: 0400
+    mode: 0644
```
Comment (maxhbr) [inline lte/gateway/deploy/roles/agw_docker/tasks/main.yml]: this contains credentials, so it should just be readable for the one that needs them.
It is generated, so it should just be readable, and not writable.

File: lte/gateway/deploy/roles/agw_docker/templates/env.j2 [hunk 2]
```diff
@@ -15,6 +15,7 @@ DOCKER_REGISTRY={{ docker_registry }}
 DOCKER_USERNAME={{ docker_user }}
 DOCKER_PASSWORD={{ docker_pass }}
 IMAGE_VERSION={{ image_version }}
+OPTIONAL_ARCH_POSTFIX=
```
Comment (maxhbr) [inline lte/gateway/deploy/roles/agw_docker/templates/env.j2]: that should be filled by ansible, right?
```suggestion
OPTIONAL_ARCH_POSTFIX={{ optional_arch_postfix }}
```

Comment (ShubhamTatvamasi) [reply lte/gateway/deploy/roles/agw_docker/templates/env.j2]: If we update this then we will have to update Ansible script for adding the value, that change will take more time.

File: lte/gateway/deploy/roles/agw_docker/tasks/main.yml (hunk 1 above)
Comment (ShubhamTatvamasi) [reply lte/gateway/deploy/roles/agw_docker/tasks/main.yml]: So if I revert this back to `0400` then I update the docs with `sudo docker-compose` commands?

File: lte/gateway/deploy/roles/agw_docker/templates/env.j2 (hunk 2 above)
Comment (maxhbr) [reply lte/gateway/deploy/roles/agw_docker/templates/env.j2]: I think that might be becoming to hacky and confusing, and it might be worth to go the clean way here. A template file that is partially filled and then largely overwritten via `sed` might be confusing and unexpected. Especially if one manually sets the repository here, having it overwritten later is bad.

Comment (ShubhamTatvamasi) [reply lte/gateway/deploy/roles/agw_docker/templates/env.j2]: Okay, I will update template code.

File: lte/gateway/deploy/roles/agw_docker/tasks/main.yml (hunk 1 above)
Comment (maxhbr) [reply lte/gateway/deploy/roles/agw_docker/tasks/main.yml]: yes, probably is better

Comment (ShubhamTatvamasi) [reply lte/gateway/deploy/roles/agw_docker/tasks/main.yml]: Okay, I will do these changes.

**Review by sebathomas — APPROVED** [review]

**Review by maxhbr — APPROVED** [review]
LGTM, for a change on the branch v1.8

File: docs/readmes/lte/deploy_install_docker.md [hunk 3]
```diff
@@ -74,10 +74,16 @@ bash agw_install_docker.sh
...
+
+```bash
+sudo -i
+```
+
 Create `control_proxy.yml` file with your orc8r details:
 
 ```bash
-cat << EOF | sudo tee /var/opt/magma/configs/control_proxy.yml
+cat << EOF | tee /var/opt/magma/configs/control_proxy.yml
```
Comment (Neudrino) [inline docs/readmes/lte/deploy_install_docker.md]: I wonder why this change would be necessary?
In the end that means that also the commands further below (l. 102ff), i.e. the `docker compose up` and `exec` would be run with elevated privileges. I would doubt that this is necessary and would rather avoid having docker run as root.

**Review by Neudrino — CHANGES_REQUESTED** [review]

File: docs/readmes/lte/deploy_install_docker.md (hunk 3 above)
Comment (ShubhamTatvamasi) [reply docs/readmes/lte/deploy_install_docker.md]: We are already becoming `root` user on the command above, so we don't need sudo commands.

Comment (maxhbr) [reply docs/readmes/lte/deploy_install_docker.md]: instead , you could have added `sudo` in front of `docker-compose` and `docker` calls

Comment (maxhbr) [reply docs/readmes/lte/deploy_install_docker.md]: > I wonder why this change would be necessary?

some of the configuration for docker compose is owned by rood and has 400 permissions

Comment (ShubhamTatvamasi) [reply docs/readmes/lte/deploy_install_docker.md]: Then we will have to run sudo with each command. That will be too much sudo. If we want that then I can update the docs.

Comment (Neudrino) [reply docs/readmes/lte/deploy_install_docker.md]: > some of the configuration for docker compose is owned by rood and has 400 permissions

Would it than not be better to allow read access to those configuration in order to _not_ run docker as root user?
I would believe getting read access to yet another config file is much less of a security concern than execution docker as root user.

Comment (ShubhamTatvamasi) [reply docs/readmes/lte/deploy_install_docker.md]: I have updated docs with sudo commands

Comment (Neudrino) [reply docs/readmes/lte/deploy_install_docker.md]: The actual question has not been answered. Executing docker as root has security implications and should not be done without a proper justification.

Comment (ShubhamTatvamasi) [reply docs/readmes/lte/deploy_install_docker.md]: @maxhbr what should we do here?

Comment (maxhbr) [reply docs/readmes/lte/deploy_install_docker.md]: >  I would believe getting read access to yet another config file is much less of a security concern than execution docker as root user.

being able to run docker is equivalent to being root. And `sudo` in front of `docker-compose` or `docker` does not elevate there permissions further.

**Review by Neudrino — APPROVED** [review]
